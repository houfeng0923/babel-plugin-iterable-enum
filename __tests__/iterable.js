import { transformAsync } from '@babel/core';
import plugin from '../src';

const options = {
  plugins: [plugin],
};

it('Transform default values', async () => {
  const input = `enum Direction { Left, Right, Down, Up };`;
  const { code } = await transformAsync(input, options);
  expect(code).toMatchSnapshot();
});


it('Transforms default string values', async () => {
  const input = `enum Direction {
    Left = 'Left',
    Right = 'Right',
    Down = 'Down',
    Up = 'Up'
  };`
  const { code } = await transformAsync(input, options);
  expect(code).toMatchSnapshot();
});

it('Transforms computed values', async () => {
  const input = `enum Direction {
    Left,
    Right,
    Down,
    Up,
    Front = Up
  };`
  const { code } = await transformAsync(input, options);
  expect(code).toMatchSnapshot();
});


it('Ignore transforms const enum', async () => {
  const input = `const enum Direction {
    Left = 'Left',
    Right = 'Right',
    Down = 'Down',
    Up = 'Up'
  };`
  const { code } = await transformAsync(input, options);
  expect(code).toMatchSnapshot();
});
