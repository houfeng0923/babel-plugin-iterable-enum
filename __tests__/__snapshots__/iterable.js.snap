// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ignore transforms const enum 1`] = `
"const enum Direction {
  Left = 'Left',
  Right = 'Right',
  Down = 'Down',
  Up = 'Up',
}
;"
`;

exports[`Transform default values 1`] = `
"const Direction = {
  Left: 0,
  Right: 1,
  Down: 2,
  Up: 3,
  \\"__proto__\\": {
    0: \\"Left\\",
    1: \\"Right\\",
    2: \\"Down\\",
    3: \\"Up\\"
  }
};
;"
`;

exports[`Transforms computed values 1`] = `
"const Direction = {
  Left: 0,
  Right: 1,
  Down: 2,
  Up: 3,
  Front: 3,
  \\"__proto__\\": {
    0: \\"Left\\",
    1: \\"Right\\",
    2: \\"Down\\",
    3: \\"Up\\",
    3: \\"Front\\"
  }
};
;"
`;

exports[`Transforms default string values 1`] = `
"const Direction = {
  Left: \\"Left\\",
  Right: \\"Right\\",
  Down: \\"Down\\",
  Up: \\"Up\\"
};
;"
`;
